openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -374,6 +379,39 @@ def run(self):
         self._quantities = []
 
         optimizer = self.options['optimizer']
+
+        # Need to tell optimizer where to put its .out files
+        if self.options['output_dir'] is None:
+            self.options['output_dir'] = "."
+        elif self.options['output_dir'] == _DEFAULT_REPORTS_DIR:
+            problem = self._problem()
+            default_output_dir = pathlib.Path(get_reports_dir()).joinpath(problem._name)
+            pathlib.Path(default_output_dir).mkdir(parents=True, exist_ok=True)
+            self.options['output_dir'] = str(default_output_dir)
```

Comment:
I'm not sure it's a good idea to be changing the value of the user facing 'output_dir' option here, since that is normally under control of the user.  Couldn't you just set the local output_dir variable, which is all that your using in subsequent code anyway?

Reply:
Correct. Fixed